### ISSUE TITLE
Unable to pass arrays inline to the constructor

### ISSUE THREAD

**Issue by anuruddhal** [ballerina-platform/ballerina-lang#13494]
**Description:**
I have a Ballerina object constructor as follows:
```ballerina
public type Definition record{
    string path;
    string method;
    !...
};

public type HTTPIngress object {
    public int port;
    public string basePath;
    public Definition[]? definitions;
    public string swaggerFilePath;

    public function __init(int port, string basePath = "", Definition[]? definitions = (), string swaggerFilePath = "")
    {
        self.port = port;
        self.basePath = basePath;
        self.definitions = definitions;
        self.swaggerFilePath = swaggerFilePath;
    }
};
```
I am trying to initialize a instance from above object definiton as below.
```ballerina
public function main(string ...args){
HTTPIngress ingress= new HTTPIngress(8080, basePath = "employee", definitions = [{
                              path: "/",
                              method: "GET"
                          }]);
}
```
I am getting following error when trying to run the program. 
```bash
error: .::employee.bal:53:100: incompatible types: expected 'Definition[]?', found 'map[]'
error: .::employee.bal:53:100: incompatible types: expected 'Definition[]?', found 'map[]'
```
However, the program compiles if I created the `Definition` array seperatly and passed it to constructor.
```ballerina
public function main(string ...args){
 Definition[] pDefintions =  [{
                              path: "/",
                              method: "GET"
                          }];
 HTTPIngress ingress= new HTTPIngress(8080, basePath = "employee", definitions = pDefintions);
}
```

**Steps to reproduce:**
- Run the program as described above.

**Affected Versions:**
0.990.0

**OS, DB, other environment details and versions:**
MacOs

**Related Issues (optional):**

**Suggested Labels (optional):**
Bug
**Suggested Assignees (optional):**

**Comment by rdhananjaya** [comment]
This seems to be fixed in 0.990.3

**Comment by hasithaa** [comment]
Verified this issue is fixed.
